### DISCUSSION TITLE
Fun Mod: Change the layout perspective

### DISCUSSION THREAD

**Discussion by Dantto** [ppy/osu#27204] [Ideas]
I was looking to do by my self a mod for osu! lazer mania just for fun.
The idea is simple: Change the perspective of the "Notes Chart" into the "Gtar Hero" or "Clone Hero" Style. (Idn the oficial name for the part of the screen where the notes fall) 

Just like this
![image](https://github.com/ppy/osu/assets/90594907/bf55b2b9-992c-435c-b4bd-ccacb9c7f8e8)

The playeable area origin is at 11/16 (little les then 2/3) on the screen.
This will Literally produce the "Mod: Hidden" or "Mod: Fade In" effect concentrate more in a smaller area and without the inconvinient of the invisible notes.

I tought i'll figure it by my own but, now i dont know even where to start or even how to change the perspective.

Why?
#1. The feeling of going fast is awesome and adictive. (Achived with perspective)
#2. Perspective is more natural.
#3. Being able to have the playable screen area tilted down feel more eficient for reading patterns
or i see it that way. If you've tried "Clone Hero" would know that, when notes appear on screen, just for some milisec they go slower and start increasing speed.

Now the intersting part is that this is a LIE. Nothing of this is really happening, is just the way our eyes feel it. 

If haven't played neither guitar nor clone hero, I know u have try the Fortnite adaptation of this.
At the end you will notice that EVERY SINGLE TIME you move from o site to another you experience this type of perspective.
Driving through the highway, cars look small and slow but they aren't.
Playing with this perspective it just feel... natural.

At this point I think of this problems:
- Change the angle of perspective of the empty playeable screen area.
- The notes start small and gets to its normal size (dictated by skin).
- Long notes tails have to go from slim to normal (from top to bottom)
-

**Comment by GSculerlor** [comment]
pretty sure I saw a WIP branch/PR for a 3D-ish perspective for the mania playfield. as far as I remember this will be implemented as a feature for mania on mobile/small-screen devices so unsure if this will be implemented as a mod.

**Comment by Akkihi** [comment]
I think this would be quite difficult to implement in the current state of the game, but having the ability to use this kind of perspective on any platform would bring several benefits:

1) Optionality – it would be an optional feature.

2) It’s genuinely convenient sometimes, because it strains the eyes less and makes you look exactly where you need to.

3) It could serve as a nice (though somewhat questionable) PR move for people who enjoy this style of gameplay. Many rhythm games with similar mechanics use exactly this perspective. If osu! were released on Steam with this kind of perspective as one of the highlighted features, it would look really cool — I often see this exact perspective in rhythm games on Steam.

However, it also creates other issues, such as skinning support, general maintenance of this perspective, and it doesn’t fully fit into osu!’s current game design (osu has always been a flat-looking game rather than a 3D/volumetric one, and that has always been both an advantage and a disadvantage).

At the very least, my comment is here because I’ve been playing significantly less lately, and at least some of that is related to:

1) The split in the community between stable and lazer.

2) In lazer, there’s a very weak sense of player-to-player communication. In stable you can just press F9 and easily chat with people, follow the chat, requests, etc.

3) General fatigue — the game itself feels tired, and what it needs isn’t just “lazer”, but a new source of excitement. (The 1v1 rating idea sounds cool, but it’s heavily limited by the small online player base and the huge skill rating spread among active players.)

In any case, I believe around 10% of mania players would definitely be happy with this kind of perspective. It would probably appeal mostly to beginners, especially since mania is often considered less interesting due to the much smaller number of available maps compared to standard mode.

I’d like to apologize for adding this extra information here — it probably deserves its own separate thread — but I still feel it’s somewhat related to the topic.
